packages/server/graphql/types/Task.ts

```diff
@@ -195,7 +196,8 @@ const Task: GraphQLObjectType = new GraphQLObjectType<any, GQLContext>({
               }
             : null
         } else if (integration.service === 'azureDevOps') {
-          const {instanceId, projectKey, issueKey} = integration
+          const {projectKey, issueKey} = integration
```

Comment:
Where are you thinking we should resolve this? 

In the other integrations, such as GitHub https://github.com/ParabolInc/parabol/blob/5e889841b96397f3c4f48769947d0f0f323b2a8b/packages/server/integrations/github/GitHubServerManager.ts#L82 and JiraServer https://github.com/ParabolInc/parabol/blob/018ab7033397fdcd6d0f09c223a3e6216d5eb847/packages/server/integrations/jiraServer/JiraServerRestManager.ts#L315, the `issueId` comes directly from the integration, e.g. GitHub or Jira. However, this may be because they provide globally unique ids, whereas Azure doesn't.

The `IntegrationHash` uses `GitHubIssueId.join`, `JiraServerIssueId.join`, or `AzureDevOpsIssueId.join`. 

A couple of related things that are confusing to me: 

1. `GitHubIssueId`, `AzureDevOpsIssueId` etc is used for the integration hash, not the issue id. Perhaps we should rename it to `AzureDevOpsIntegrationHash`?
2. We're using the `projectId` and `projectKey` interchangeably, e.g. https://github.com/ParabolInc/parabol/blob/62c43cf831b63eeff96905c2a4330ca8dddfe879/packages/server/graphql/types/EstimateStage.ts#L137. It seems that we should remove `projectKey` and just use `projectId`. 

I'm happy to create issues for these if you agree.

Reply:
Makes sense! Looks like that would work in a similar way to JiraIssue: https://github.com/ParabolInc/parabol/blob/36b1cd310528895db60e9d1d3be2f5d03ddf6c10/packages/server/graphql/types/JiraIssue.ts#L23

Reply:
A few notes after looking into this more:

1. An id of `instanceId:issueKey` wouldn't have been unique as the `instanceId` can be `dev.azure.com/ParabolDevelopment` and the `issueKey` can be 7. We need the `projectId` to make it unique
2. `teamProject` is defined as the name here https://github.com/ParabolInc/parabol/blob/26ec310dcc9e2b35be2aad5922da62aa7cc5ea59/packages/server/graphql/types/AzureDevOpsWorkItem.ts#L25 but it can be a team project id
3. To make the `AzureDevOpsWorkItem` id unique, we can do:

```
    id: {
      type: new GraphQLNonNull(GraphQLID),
      description: 'GUID instanceId:projectKey:issueKey',
      resolve: ({id, teamProject, url}: {id: string; teamProject: string; url: string}) => {
        const instanceId = getInstanceId(new URL(url))
        return AzureDevOpsIssueId.join(instanceId, teamProject, id)
      }
    }
```
4. The work item id is being used as a key elsewhere in the app, e.g. https://github.com/ParabolInc/parabol/blob/04dd285132de56059a30142f4e0b69991ff56645/packages/client/components/AzureDevOpsScopingSearchResultItem.tsx#L104. We need to update these references.


I've created this issue to update the id and handle the changes elsewhere in the app: https://github.com/ParabolInc/parabol/issues/7038

Reply:
> We need the projectId to make it unique

yep!

> AzureDevOpsWorkItem

is teamProject the same as projectId? If so, I'd opt for `projectId` everywhere in _our_ codebase since the ID suffix makes it clear it's a string. Although i don't really care either way, as long as the name in consistent.

> he work item id is being used as a key elsewhere in the app

agreed, let's not use our own internal UID for anything user-facing

Reply:
Great! I've created an issue to refactor `teamProject` as it looks like it's being used as both a `name` and an `id`: https://github.com/ParabolInc/parabol/issues/7073